core/src/main/java/org/kabeja/entities/Arc.java

```diff
@@ -241,4 +260,43 @@ public void setCounterClockwise(boolean counterclockwise) {
     public void transform(TransformContext context) {
       this.center = context.transform(this.center);
     }
+
+    @Override
+    public void toWcs() {
+        Extrusion e = this.getExtrusion();
+        if (getExtrusion().getNormal().equals(new Vector(0,0,1))) {
+            return;
+        }
+        Point3D transformedStart = e.transformOcsToWcs(getPointInOcs(this.start_angle));
+        Point3D transformedEnd = e.transformOcsToWcs(getPointInOcs(this.end_angle));
+        this.center = e.transformOcsToWcs(this.center);
+
+        Extrusion newE = new Extrusion();
+        newE.setX(0);
+        newE.setY(0);
+        newE.setZ(1);
+
+        transformedStart = newE.wcsToOcs(transformedStart);
+        transformedEnd = newE.wcsToOcs(transformedEnd);
+        this.start_angle = getAngleAtPoint(transformedStart);
+        this.end_angle = getAngleAtPoint(transformedEnd);
+
+        // @ToDo is this the general criterium?
+        if (e.getZ() < 0) {
+            counterclockwise = !counterclockwise;
+        }
+    }
+
+    /**
+     *
+     * @param p Point in OCS, should be on the arc
```

Comment:
What happens if it isn't?

Reply:
I think that in that case it returns something that doesn't make a lot of sense, since the calculation uses the radius to determine the angle.
But now that I think about it I think I could use distance(centerpoint, p) instead of the radius in oder to get usable results no matter what. Should I do that?

Reply:
discussion: check if point is actually on circle, fail if not